static/app/components/assistant/guideAnchor.tsx

```diff
@@ -33,216 +33,180 @@ type Props = {
     pathname: string;
     query: Query;
   };
-};
-
-type State = {
-  active: boolean;
-  orgId: string | null;
-  step: number;
-  currentGuide?: Guide;
-};
-
-class BaseGuideAnchor extends Component<Props, State> {
-  state: State = {
-    active: false,
-    step: 0,
-    orgId: null,
-  };
-
-  componentDidMount() {
-    const {target} = this.props;
-    target && registerAnchor(target);
-  }
-
-  componentDidUpdate(_prevProps: Props, prevState: State) {
-    if (this.containerElement.current && !prevState.active && this.state.active) {
-      try {
-        const {top} = this.containerElement.current.getBoundingClientRect();
-        const scrollTop = window.pageYOffset;
-        const centerElement = top + scrollTop - window.innerHeight / 2;
-        window.scrollTo({top: centerElement});
-      } catch (err) {
-        Sentry.captureException(err);
-      }
+}>;
+
+function BaseGuideAnchor({
+  target,
+  onFinish,
+  children,
+  position,
+  offset,
+  containerClassName,
+  to,
+}: Props) {
+  const guideContainerRef = useRef<HTMLSpanElement>(null);
+  const {orgId, forceHide, currentGuide, currentStep} = useLegacyStore(GuideStore);
+
+  const isActive =
+    forceHide !== true && currentGuide?.steps[currentStep]?.target === (target ?? false);
```

Comment:
Can drop the nullish coalescing and exit early if no target is present to make it a bit easier to read? Maybe something like
```suggestion
  const isActive =
    !forceHide || target && currentGuide?.steps[currentStep]?.target === target;
```